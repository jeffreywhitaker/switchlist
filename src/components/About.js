// import dependencies
import React from 'react'
import styled from 'styled-components'

// About page component
export default function About() {
    return (
        <AboutWrapper>
            <h2>Hi, I'm Jeff. Nice to meet you!</h2>
            <p>I'm a fullstack web developer in training. Currently a student at Lambda School, this is my first private side project. Thank you for checking in!</p> 
        </AboutWrapper>
    )
}

// styled components
const AboutWrapper = styled.section`
  width: 65%
  margin: 0 auto
`