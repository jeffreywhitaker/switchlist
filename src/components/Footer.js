// import dependencies
import React from 'react'
import styled from 'styled-components'

// Footer component
export default function Footer() {
    return (
        <FooterSection>
            <p>SwitchList</p>
            <p>Copyright Jeffrey David Whitaker, 2019</p>
        </FooterSection>
    )
}

// styled components
const FooterSection = styled.section`
    border-top: 1px solid black
    text-align: center
    margin-top: 10px
`