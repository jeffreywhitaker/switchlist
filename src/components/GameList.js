// import dependencies
import React, { useEffect } from 'react'
import { connect } from 'react-redux'
import styled from 'styled-components'

// import components and functions
import GameCard from './GameCard'
import { getGames } from '../actions/actions'

// GameList component
function GameList({ games, getGames }) {
  useEffect(() => {
    getGames()
  }, [getGames])

  return (
    <GameListSection>
      <ResultsPerPageDiv>
        <span>Results per page</span>
        <select name="results per page">
          <option>10</option>
          <option>25</option>
          <option>50</option>
          <option>100</option>
        </select>
      </ResultsPerPageDiv>
      {games.list.map((game) => {
        return <GameCard game={game} key={game.name} />
      })}
    </GameListSection>
  )
}

// connect to Redux state
const mapStateToProps = (state) => {
  return {
    games: state.games,
  }
}

// export component
export default connect(mapStateToProps, { getGames })(GameList)

// styled components

const GameListSection = styled.section``

const ResultsPerPageDiv = styled.div`
    display: flex
    justify-content: flex-end
    margin-right: 30px
`
