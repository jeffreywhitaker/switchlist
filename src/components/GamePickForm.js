// import dependencies
import React, { useState, useEffect } from 'react'
import { connect } from 'react-redux'
import styled from 'styled-components'

// import functions
import { getDataList, getGames } from '../actions/actions'

// GamePickForm component
function GamePickForm({ publishers, composers, getDataList, getGames }) {
  useEffect(() => {
    getGames()
    getDataList()
  }, [getGames, getDataList])

  const defaultFilter = {
    title: '',
    series: '',
    publisherId: '',
    composerId: '',
    hdRumble: false,
    multiplayer: false,
    cloudSaves: false,
    genre: '',
  }

  const [filterData, setFilterData] = useState(defaultFilter)

  // reload games with filter data
  const handleFilter = (e) => {
    e.preventDefault()
    console.log('filterdata', filterData)

    let filterDataToSend = {}
    // for each key in filterData
    for (const key in filterData) {
      if (filterData[key] !== defaultFilter[key]) {
        filterDataToSend[key] = filterData[key]
      }
    }
    getGames(filterDataToSend)
  }

  const handleValueChange = (e) => {
    const value =
      e.target.type === 'checkbox' ? e.target.checked : e.target.value
    setFilterData({
      ...filterData,
      [e.target.name]: value,
    })
    console.log('e target name/value', e.target.name, e.target.value)
  }

  return (
    <GamePickFormSection onSubmit={handleFilter}>
      <ColumnDiv1>
        <h3>Search by</h3>
        <button type="submit">Search</button>
      </ColumnDiv1>
      <ColumnDiv>
        <GPFsubDiv>
          <p>Title</p>
          <input
            type="text"
            name="title"
            placeholder="all or part of title"
            value={filterData.title}
            onChange={handleValueChange}
          />
        </GPFsubDiv>
        <GPFsubDiv>
          <p>Publisher</p>
          {/* <input
            type="text"
            name="publisher"
            value={filterData.publisher}
            onChange={handleValueChange}
          /> */}
          <select
            name="publisherId"
            placeholder="publisher"
            value={filterData.publisherId}
            onChange={handleValueChange}
            size={1}
          >
            <option value=""></option>
            {publishers.map((publisher) => {
              return (
                <option value={publisher._id} key={publisher.name}>
                  {publisher.name}
                </option>
              )
            })}
          </select>
        </GPFsubDiv>
        <GPFsubDiv>
          <p>Series</p>
          <input
            type="text"
            name="series"
            value={filterData.series}
            onChange={handleValueChange}
          ></input>
        </GPFsubDiv>
        <GPFsubDiv>
          <p>Composers</p>
          <select
            name="composerId"
            placeholder="composer"
            value={filterData.composerId}
            onChange={handleValueChange}
            size={1}
          >
            <option value=""></option>
            {composers.map((composer) => {
              return (
                <option value={composer._id} key={composer.name}>
                  {composer.name}
                </option>
              )
            })}
          </select>
        </GPFsubDiv>
      </ColumnDiv>
      <ColumnDiv>
        <GPFsubDiv>
          <p>HD rumble</p>
          <input
            type="checkbox"
            name="hdRumble"
            value={filterData.hdRumble}
            onChange={handleValueChange}
          ></input>
        </GPFsubDiv>
        <GPFsubDiv>
          <p>Multiplayer</p>
          <input
            type="checkbox"
            name="multiplayer"
            value={filterData.multiplayer}
            onChange={handleValueChange}
          ></input>
        </GPFsubDiv>
        <GPFsubDiv>
          <p>Cloud Saves</p>
          <input
            type="checkbox"
            name="cloudSaves"
            value={filterData.cloudSaves}
            onChange={handleValueChange}
          ></input>
        </GPFsubDiv>
      </ColumnDiv>
      <ColumnDiv>
        <GPFsubDiv>
          <p>Genres</p>
          <input
            type="text"
            name="genre"
            value={filterData.genre}
            onChange={handleValueChange}
          ></input>
        </GPFsubDiv>
      </ColumnDiv>
    </GamePickFormSection>
  )
}

// connect to Redux state
const mapStateToProps = (state) => {
  return {
    publishers: state.datalist.publishers,
    composers: state.datalist.composers,
  }
}

// export component
export default connect(mapStateToProps, { getDataList, getGames })(GamePickForm)

// styled components

const GamePickFormSection = styled.form`
  display: flex
  flex-wrap: wrap
  justify-content: space-around
  align-items: baseline
  border-top: 1px solid black
  border-bottom: 1px solid black
  margin: 10px 20px
  padding: 20px 30px
`

const GPFsubDiv = styled.div`
  margin: 10px;
`

const ColumnDiv = styled.div`
  display: flex
  flex-direction: column
`

const ColumnDiv1 = styled.div`
  height: 100%
  display: flex
  flex-direction: column
  justify-content: space-between
`
