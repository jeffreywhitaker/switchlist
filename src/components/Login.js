import React, { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import styled from 'styled-components'

import { userLogin } from '../actions/actions'

function Login({ userLogin, history, login }) {
    const [credentials, setCredentials] = useState({username: '', password: ''})

    const callLogin = e => {
        e.preventDefault()
        userLogin(credentials)
    }

    useEffect(() => {
        if (login.isSuccessful) {history.push('/gamelist')}
    }, [login.isSuccessful, history])

    const handleValueChange = e => {
        setCredentials( {
            ...credentials,
            [e.target.name]: e.target.value,
        })
    }

    return (
        <LoginDiv>
            <form onSubmit={callLogin}>
                <p>Username:</p>
                <input
                    type="text"
                    name="username"
                    value={credentials.username}
                    onChange={handleValueChange}
                />
                <p>Password:</p>
                <input
                    type="password"
                    name="password"
                    value={credentials.password}
                    onChange={handleValueChange}
                />
                <br/>
                <br/>
                <button>Log in</button>
            </form>
            <h2>Not registered?</h2>
            <Link to="/signup" >Sign Up!</Link>
        </LoginDiv>
    )
}

const mapStateToProps = state => {
    return {
        login: state.login
    }
  }
  
  export default connect(mapStateToProps, { userLogin })(Login)

  // styled components

  const LoginDiv = styled.div`
  width: 200px
  margin: 20px auto
`