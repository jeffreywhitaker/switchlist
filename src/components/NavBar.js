import React from 'react'
import { NavLink } from 'react-router-dom'
import styled from 'styled-components'


function NavBar() {
  return (
    <NavHeader>
        <LogoHeader>SwitchList</LogoHeader>
        <nav>
            <StyledNavLink to={'/'}>Home</StyledNavLink>
            <StyledNavLink to={'/gamelist'}>Game List</StyledNavLink>
            <StyledNavLink to={'/about'}>About</StyledNavLink>
            <StyledNavLink to={'/login'}>Login / Signup</StyledNavLink>
        </nav>
    </NavHeader>
  )
}

export default NavBar

// styled components

const NavHeader = styled.header`
    border: 1px solid black
    display: flex
    justify-content: space-around
    align-items: center
`

const LogoHeader = styled.h1`
  font-family: calibri
  font-size: 30px
`


const StyledNavLink = styled(NavLink)`
    text-decoration: none
    padding: 20px
    background: cyan
    border: 1px solid black
    margin: 10px 15px
`