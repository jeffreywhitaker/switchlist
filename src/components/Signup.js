import React, { useState } from 'react'
import { connect } from 'react-redux'
import styled from 'styled-components'

import { postUser } from '../actions/actions'

const initialCredentials = {
    username: '',
    password: '',
    confirmPassword: ''
}

function Signup({ postUser, history }) {

    const [credentials, setCredentials] = useState(initialCredentials)

    function handleChange(e) {
        setCredentials(
            {
                ...credentials,
                [e.target.name]: e.target.value
            }
        )
    }

    function callSignUp(e) {
        e.preventDefault()
        postUser(credentials)
        history.push('/gamelist')
    }

    return (
        <SignUpDiv>
            <br/>
            <br/>
            <form onSubmit={callSignUp}>
                <input
                    type="text"
                    name="username"
                    value={credentials.username}
                    placeholder="desired user name"
                    onChange={handleChange}
                />
                <input
                    type="password"
                    name="password"
                    value={credentials.password}
                    placeholder="password"
                    onChange={handleChange}
                />
                <input
                    type="password"
                    name="confirmPassword"
                    value={credentials.confirmPassword}
                    placeholder='confirm password'
                    onChange={handleChange}
                />
                <button>Sign Up</button>
            </form>
        </SignUpDiv>
    )
}

const mapStateToProps = state => {
    return {
        signup: state.signup
    }
  }
  
  export default connect(mapStateToProps, { postUser })(Signup)

//styled components

const SignUpDiv = styled.div`
  width: 200px
  margin: 20px auto
`