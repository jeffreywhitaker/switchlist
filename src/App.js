// import dependencies
import React from 'react'
import { Route } from 'react-router-dom'
import styled from 'styled-components'

// import components
import NavBar from './components/NavBar'
import Home from './components/HomePage'
import GamePickForm from './components/GamePickForm'
import GameList from './components/GameList'
import About from './components/About'
import Login from './components/Login'
import Signup from './components/Signup'
import UserProfile from './components/UserProfile'
import Footer from './components/Footer'
import PrivateRoute from './components/PrivateRoute'

// App component
export default function App() {
  return (
    <AppWrapper className="appWrapper">
      <Route path="/" component={NavBar} />
      <Route exact path="/" component={Home} />
      <Route path="/gamelist" component={GamePickForm} />
      <Route path="/gamelist" component={GameList} />
      <Route path="/about" component={About} />
      <Route path="/login" component={Login} />
      <Route path="/signup" component={Signup} />
      <PrivateRoute path="/profile" component={UserProfile} />
      <Footer />
    </AppWrapper>
  )
}

// styled components
const AppWrapper = styled.div`
  max-width: 1100px
  margin: 0 auto
`
