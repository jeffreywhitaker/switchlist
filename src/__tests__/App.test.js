import React from 'react'
import { Route } from 'react-router-dom'
import styled from 'styled-components'
import ShallowRenderer from 'react-test-renderer/shallow'
import App from '../App'

// import components
import NavBar from '../components/NavBar'
import Home from '../components/HomePage'
import GamePickForm from '../components/GamePickForm'
import GameList from '../components/GameList'
import About from '../components/About'
import Login from '../components/Login'
import Signup from '../components/Signup'
import UserProfile from '../components/UserProfile'
import Footer from '../components/Footer'
import PrivateRoute from '../components/PrivateRoute'

describe('the App component', () => {
  it('matches previous snapshot', () => {
    const renderer = new ShallowRenderer()
    renderer.render(<App />)
    const output = renderer.getRenderOutput()
    expect(output).toMatchSnapshot()
  })

  // it('renders the proper child components', () => {
  //   const renderer = new ShallowRenderer()
  //   renderer.render(<App />)
  //   const output = renderer.getRenderOutput()
  //   expect(output.props.children).toEqual([
  //     <AppWrapper className="appWrapper">
  //       <Route path="/" component={NavBar} />
  //       <Route exact path="/" component={Home} />
  //       <Route path="/gamelist" component={GamePickForm} />
  //       <Route path="/gamelist" component={GameList} />
  //       <Route path="/about" component={About} />
  //       <Route path="/login" component={Login} />
  //       <Route path="/signup" component={Signup} />
  //       <PrivateRoute path="/profile" component={UserProfile} />
  //       <Footer />
  //     </AppWrapper>
  //   ])
  // })
})

// styled components
const AppWrapper = styled.div`
  max-width: 1100px
  margin: 0 auto
`
