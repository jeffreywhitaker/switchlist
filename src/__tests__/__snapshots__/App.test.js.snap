// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the App component matches previous snapshot 1`] = `
<ForwardRef(styled.div)
  className="appWrapper"
>
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(NavBar)",
        "type": [Function],
      }
    }
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(GamePickForm)",
        "type": [Function],
      }
    }
    path="/gamelist"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(GameList)",
        "type": [Function],
      }
    }
    path="/gamelist"
  />
  <Route
    component={[Function]}
    path="/about"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Login)",
        "type": [Function],
      }
    }
    path="/login"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Signup)",
        "type": [Function],
      }
    }
    path="/signup"
  />
  <PrivateRoute
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(UserProfile)",
        "type": [Function],
      }
    }
    path="/profile"
  />
  <Footer />
</ForwardRef(styled.div)>
`;
